### DISCUSSION TITLE
Configuring RabbitMQ user and password

### DISCUSSION THREAD

**Discussion by alex-todorov-j2** [microsoft/aspire#3560] [Q&A] [answered]
In Program of AppHost if I add a RMQ dependency:

 ```
var msgBroker = builder.AddRabbitMQ("messaging");   
var apiService = builder.AddProject<Projects.AspireStarterApp_ApiService>("apiservice")
    .WithReference(msgBroker);
```
and then go to the /health endpoint for _apiService_ in the browser, I'm getting a Healthy response.

But if I change the code above to use user/psw I configure:

```
var msgBroker = builder.AddRabbitMQ("messaging")
    .WithEnvironment("RABBITMQ_DEFAULT_USER", "admin")
    .WithEnvironment("RABBITMQ_DEFAULT_PASS", "7da0f3e209814fe89d60603192c0dc3b");
```
then RMQ logs _PLAIN login refused: user 'guest'_ and of course /health reports Unhealthy.

My questions:
- Where else do I have to configure the RMQ connection in the .NET Aspire solution?
- Is there a complete example with apsettings.json or other config that I can use to get RMQ IConnection ready to inject in app code?

**Comment by timheuer** [comment, marked as answer]
Use parameters and the ctor of the `AddRabbitMQ` method:

```csharp
var rabbitUser = builder.AddParameter("RabbitUser");
var rabbitPass = builder.AddParameter("RabbitPassword", true);

var msgBroker = builder.AddRabbitMQ("messaging", rabbitUser, rabbitPass);
```

using this method would be looking for Configuration value of `Parameters:RabbitUser` and such (could put in user secrets, appsettings)

**Reply by alex-todorov-j2** [reply]
Yes, it worked, after updating VS Preview to latest. 

The WithEnvironment still doesn't work for reasons that elude me at the moment.

**Reply by aidasofialily-cmd** [reply]
Yes it works, after VS Code in completed.

The environment work for the reasons for a moment.

**Comment by alex-todorov-j2** [comment]
Thanks but... builder is IDistributedApplicationBuilder in the AppHost Program and WebApplicationBuilder in the ApiService. Neither has an AddParameter method. Time to update my VS Preview, I guess. Hopefully that should give me the latest .net aspire.

**Reply by davidfowl** [reply]
Yes you should get the latest version of aspire.

**Comment by inlineHamed** [comment]
This worked for me :
```
var rabbitMq = builder.AddRabbitMQ("messaging",
              userName: builder.AddParameter("username", "admin", secret: true),
              password: builder.AddParameter("password", "123456", secret: true))
       .WithManagementPlugin(port: 8002);
```
aspire dotnet 9.2.1

**Comment by aidasofialily-cmd** [comment]
@dotnet What is aspire dotnet 9.2.1

`var rabbitMq = builder.AddRabbitMQ("messaging",
              userName: builder.AddParameter("username", "admin", secret: true),
              password: builder.AddParameter("password", "123456", secret: true))
       .WithManagementPlugin(port: 8002`
